src/representations/representation_manager_spherical_covariants.hh

```diff
@@ -0,0 +1,506 @@
+        internal::SphericalCovariantsPrecomputation<Type>>(
+        spherical_covariants_precompute);
+  }
+
+  template <class StructureManager>
+  class RepresentationManagerSphericalCovariants
+      : public RepresentationManagerBase {
+   public:
+    using Manager_t = StructureManager;
+    using ManagerPtr_t = std::shared_ptr<Manager_t>;
+    using Hypers_t = RepresentationManagerBase::Hypers_t;
+    using Key_t = std::vector<int>;
+    using SparseProperty_t =
+        BlockSparseProperty<double, 1, 0, Manager_t, Key_t>;
+    using Data_t = typename SparseProperty_t::Data_t;
+
+    RepresentationManagerSphericalCovariants(ManagerPtr_t sm,
+                                             const Hypers_t & hyper)
+        : soap_vectors{*sm}, structure_manager{sm}, rep_expansion{std::move(sm),
+                                                                  hyper} {
+      this->set_hyperparameters(hyper);
+    }
+
+    //! Copy constructor
+    RepresentationManagerSphericalCovariants(
+        const RepresentationManagerSphericalCovariants & other) = delete;
+
+    //! Move constructor
+    RepresentationManagerSphericalCovariants(
+        RepresentationManagerSphericalCovariants && other) = default;
+
+    //! Destructor
+    virtual ~RepresentationManagerSphericalCovariants() = default;
+
+    //! Copy assignment operator
+    RepresentationManagerSphericalCovariants &
+    operator=(const RepresentationManagerSphericalCovariants & other) = delete;
+
+    //! Move assignment operator
+    RepresentationManagerSphericalCovariants &
+    operator=(RepresentationManagerSphericalCovariants && other) = default;
+
+    void set_hyperparameters(const Hypers_t & hypers) {
+      using internal::SphericalCovariantsType;
+      this->max_radial = hypers.at("max_radial").get<size_t>();
+      this->max_angular = hypers.at("max_angular").get<size_t>();
+      this->spherical_covariants_type_str =
+          hypers.at("soap_type").get<std::string>();
+      this->lambda = hypers.at("lam").get<size_t>();
+      this->inversion_symmetry = hypers.at("inversion_symmetry").get<bool>();
+      this->normalize = hypers.at("normalize").get<bool>();
+
+      if (this->spherical_covariants_type_str.compare("LambdaSpectrum") == 0) {
+        this->spherical_covariants_type =
+            SphericalCovariantsType::LambdaSpectrum;
+        this->precompute_spherical_covariants[enumValue(
+            SphericalCovariantsType::LambdaSpectrum)] =
+            make_spherical_covariants_precompute<
+                SphericalCovariantsType::LambdaSpectrum>(hypers);
+
+      } else {
+        throw std::logic_error("Requested Spherical Covariants type \'" +
+                               this->spherical_covariants_type_str +
+                               "\' has not been implemented.  Must be one of" +
+                               ": \'LambdaSpectrum\'.");
+      }
+    }
+
+    std::vector<Precision_t> & get_representation_raw_data() {
+      return this->dummy;
```

Comment:
I guess this is just a stub and will at some point be implemented? Or is it either or with the sparse raw data return below?
Will this be a function which gives a non-sparse view on the sparse data?

Reply:
this will disapear

Reply:
Added a comment to clarify that it's an unused function and you should never call it